provisioning/baremetal_pod.go

```diff
@@ -471,7 +471,9 @@ func NewMetal3Deployment(targetNamespace string, images *Images, config *metal3i
 func GetExistingMetal3Deployment(client appsclientv1.DeploymentsGetter, targetNamespace string) (bool, error) {
 	existing, err := client.Deployments(targetNamespace).Get(context.Background(), baremetalDeploymentName, metav1.GetOptions{})
 	if existing != nil && err == nil {
-		return true, nil
+		if _, maoOwned := existing.Annotations["machine.openshift.io/owned"]; maoOwned {
+			return true, nil
+		}
```

Comment:
i think this changes the entire meaning of the function from does it exist, to is the deployment owned by mao
maybe keep it out of the function?

Reply:
That does sounds less hacky, but I think we have to follow the plan because we need the MAO to do the right thing during downgrades, and the thing we taught it to do in 4.6 is look at the annotations. As far as I know the MAO doesn't update the Deployment at all once it exists, so I doubt that it rewrites the ControllerRef on downgrade, even if it does set it on create.
Again, this code is going away soon anyway. I _think_ we could actually just delete it now and it would work, since we call ApplyDeployment() and IIUC that should rewrite the annotations. But Sandhya has improved tests and stuff in #65, so I'm leaving it for that PR.

Reply:
MAO adds the maoOwnedAnnotations to all the resources it creates and so we just using it.

This documentation about ControllerRef says it is partially implemented https://stupefied-goodall-e282f7.netlify.app/contributors/design-proposals/api-machinery/controller-ref/. 

This is hand-off mechanism is required only during upgrade/downgrade between 4.6 and 4.7.

Reply:
I do not think the name suggested here is correct. We want it to be a generic method to see if a metal3 deployment already exists after/during an upgrade. When we upgrade from 4.6 to 4.7, the older metal3 deployment would have been created by MAO. From 4.7 to 4.8, it would have been created by CBO. 
So, I am not for adding MAO in the name of the method.

Reply:
I am expecting that this method will either go away or be completely rewritten (with a new name) in #65.